22.md

```diff
@@ -0,0 +1,58 @@
+LUD-22: Webhook notification support in `payRequest`.
+=================================
+
+`author: andreneves` `author: fiatjaf`
+
+---
+
+## Support for LNURL-Pay webhook notification
+
+This allows for a `WALLET` to pass a custom `webhook` property to a `SERVICE` that accepts it. This can have multiple use-cases. The idea is to allow a `WALLET` to request for the `SERVICE` to send updates on whether a given invoice has been settled to a 3rd party `SERVICE` or `PROVIDER`.
+
+This is completely additional/ad-hoc, so current implementations are not affected by this. Any services and wallets that wish to support it do need to add some more business logic.
+
+### Service-side
+
+`SERVICE` must alter its JSON response to the first callback to include a `webhookAllowed` field, as follows:
+
+```diff
+ {
+   callback: String,
+   maxSendable: MilliSatoshi,
+   minSendable: MilliSatoshi,
+   metadata: String,
++  webhookAllowed: Boolean,
+   tag: "payRequest",
+ }
+```
+
+The value of `webhookAllowed` should be of boolean type -- `true` or `false`. `WALLET` should interpret the property as `false` if is not provided on the response payload.
+
+### Wallet-side
+
+Upon seeing a `webhookAllowed` property set to `true` on the response from the `SERVICE`, a `WALLET` may then pass along a new query parameter named `webhook` to the second callback to `SERVICE`:
+
+```diff
+- <callback><?|&>amount=<milliSatoshi>
++ <callback><?|&>amount=<milliSatoshi>&webhook=<String>
+```
+
+### Note on Webhook URL parameters
+
+Webhook URLs must be SSL-enabled `https` URLs, and the `SERVICE` is encouraged to enable any and all restrictions and safety checks on their side. Additionally, [GET URL's accept around ~2000 characters for the entire request string](https://stackoverflow.com/a/417184). Therefore `webhook` query parameter can only be as large as to fit in the URL alongside any/all of the other properties outlined above.
```

Comment:
nit: Perhaps allow http on Onion service.

The GET URL limit kinda applies to every parameter and could be a notice in LUD-06 instead.